WordPress/Classes/ViewRelated/Gutenberg/Layout Picker/GutenbergLayoutPickerViewController.swift

```diff
@@ -2,11 +2,20 @@ import UIKit
 import Gridicons
 import Gutenberg
 
-class GutenbergLayoutSection {
+class GutenbergLayoutSection: CategorySection {
     var section: PageTemplateCategory
     var layouts: [PageTemplateLayout]
     var scrollOffset: CGPoint
 
+    var title: String { section.title }
+    var emoji: String? { section.emoji }
+    var categorySlug: String { section.slug }
+    var description: String? { section.desc }
+    var thumbnails: [Thumbnail] {
+        // TODO: pass device different modes after data model supports them
+        layouts.map { Thumbnail(urlDesktop: $0.preview, urlTablet: $0.preview, urlMobile: $0.preview, slug: $0.slug) }
```

Comment:
I meant to leave a comment here when I pushed the draft. I originally tried the protocol approach you mentioned, but was surprised that it would not work the way we want, since `var thumbnails: [Thumbnail] { get }` is required by the adopted `CategorySection` protocol in the subclasses' collections. Specifically, what I ran into was that Swift does not support conformance to read-only property requirements as a covariant ([this answer on SO](https://stackoverflow.com/a/42716340) was informative).

I used this computed mapping as a stop-gap for now, but I'm still looking for a better pattern to improve this.

Reply:
Ah, I see the issue you were running into, I didn't consider that limitation. I wonder if it would still be beneficial to set up a protocol and use the `thumbnails` getter as a wrapper for `layouts`. I'm thinking of something like https://github.com/wordpress-mobile/WordPress-iOS/commit/d50ba9ad24337b81d0ad00970f8b2cf079d561eb

To be clear though I thinking the effort of mapping to a struct is minimal so not a show stopper.

Reply:
Nice, thank you for the suggestion! I pulled the changes from your branch to make it a protocol and use the wrapper property in the implementation. :+1:

Reply:
The only thing that we lose is the namespacing for Thumbnail, but I don't think that's a huge issue, and we could prepend it in the future if the need arises.